cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/service/DraftHandler.java

```diff
@@ -0,0 +1,215 @@
+
+      DraftId id = new DraftId(namespace, draftId, request.getUserId());
+      draftService.writeDraft(id, draftStoreRequest);
+
+      responder.sendStatus(HttpURLConnection.HTTP_OK);
+    });
+  }
+
+  /**
+   * Deletes a draft
+   */
+  @DELETE
+  @Path("v1/contexts/{context}/drafts/{draft}")
+  public void deleteDraft(HttpServiceRequest request, HttpServiceResponder responder,
+                          @PathParam("context") String namespaceName,
+                          @PathParam("draft") String draftId) {
+    respond(namespaceName, responder, (namespace) -> {
+      DraftId id = new DraftId(namespace, draftId, request.getUserId());
+
+      draftService.deleteDraft(id);
+      responder.sendStatus(HttpURLConnection.HTTP_OK);
+    });
+  }
+
+  /**
+   * Utility method to correct deserialize the config field in the {@link DraftStoreRequest} object
+   * @param jsonStr the json string representing the DraftStoreRequest
+   * @return {@link DraftStoreRequest} object
+   */
+  private DraftStoreRequest<ETLConfig> deserializeDraftStoreRequest(String jsonStr) {
+    try {
+      DraftStoreRequest<ETLConfig> request = GSON
+        .fromJson(jsonStr, new TypeToken<DraftStoreRequest<ETLConfig>>() { }.getType());
+
+      if (request.getArtifact() == null) {
+        throw new IllegalArgumentException("artifact is null");
+      }
+
+      if (StudioUtil.isBatchPipeline(request.getArtifact())) {
+        return GSON.fromJson(jsonStr, new TypeToken<DraftStoreRequest<ETLBatchConfig>>() { }.getType());
+      } else if (StudioUtil.isStreamingPipeline(request.getArtifact())) {
+        return GSON.fromJson(jsonStr, new TypeToken<DraftStoreRequest<DataStreamsConfig>>() { }.getType());
+      } else {
+        throw new IllegalArgumentException(String
+                                             .format("artifact '%s' is not supported, valid options are: '%s' or '%s'",
+                                                     request.getArtifact().getName(), StudioUtil.ARTIFACT_BATCH_NAME,
+                                                     StudioUtil.ARTIFACT_STREAMING_NAME));
+      }
+    } catch (JsonSyntaxException e) {
+      throw new IllegalArgumentException("Unable to decode request body: " + e.getMessage());
+    } catch (IllegalArgumentException e) {
+      throw new IllegalArgumentException("Invalid config: " + e.getMessage());
+    }
+  }
+
+  /**
+   * Utility method that checks that the namespace exists and fetches current user before responding.
+   */
+  private void respond(String namespaceName, HttpServiceResponder responder, NamespacedEndpoint endpoint) {
+    SystemHttpServiceContext context = getContext();
+    NamespaceSummary namespaceSummary;
+    try {
+      namespaceSummary = context.getAdmin().getNamespaceSummary(namespaceName);
+      if (namespaceSummary == null) {
+        responder.sendError(HttpURLConnection.HTTP_NOT_FOUND, String.format("Namespace '%s' not found", namespaceName));
+        return;
```

Comment:
Seems not needed?

Reply:
This is needed so that we exit if the namespace is not found, we cant proceed without a valid namespace